In [6]:
import numpy as np
import statsmodels.stats.proportion as smp

### Задание 1

- Вы провели эксперимент c упрощением формы заказа в магазине Утконос и получили результаты по метрике конверсий в покупку. Выберите метод оценки и оцените есть ли стат.значимые различия между конверсиями в двух группах при alpha = 5%. Дайте краткие рекомендации команде.
        Результаты: 
1.  Число юзеров в группах , которые заходили на сайт в период эксперимента: n1 = 15550 и n2 = 15550 . 
2.  Число юзеров в группах , которые совершили хотя бы одну покупку за период эксперимента:  n1 = 164 и n2 = 228 
3.  Конверсии : conv1 =  1.05%  conv2 =  1.47% . 


In [7]:
alpha = 0.05 # число успехов в каждой группе
successes = np.array([164, 228])

In [8]:
# число наблюдений в каждой группе
nobs = np.array([15550, 15550])

In [9]:
# пропорции успехов в каждой группе
proportions = successes / nobs

In [10]:
# z-статистика и p-значение
zstat, pvalue = smp.proportions_ztest(count=successes, nobs=nobs, value=0, alternative='two-sided')
print(zstat)
print(pvalue)

-3.253054722696254
0.001141715077067056


На основе z-статистики: -3.253 и p-value: 0.001, можно сделать вывод о статистически значимых различиях между конверсиями в двух группах.

Рекомендации: на основании результатов эксперимента можно рекомендовать использовать упрощенную форму заказа в магазине Утконос, так как она приводит к более высокой конверсии в покупку. Однако, следует провести дополнительные исследования, чтобы понять, какие именно элементы формы заказа необходимо упростить, чтобы максимизировать конверсии.

### Задание 2

Сравниваем метрику конверсия в покупку. Размер выборки - 10000 элементов в каждой группе . Какой статистический критерий тут лучше всего подойдёт и почему ?

Для сравнения метрики конверсии в покупку между двумя группами с размером выборки по 10 000 элементов в каждой группе лучше всего использовать z-тест, так как объём выборки достаточно большой (больше 30), что позволяет использовать нормальное распределение для вычисления статистической значимости. T-тест Стьюдента также может использоваться в данном случае, но z-тест является более точным и надёжным при больших размерах выборки.